### PR TITLE
Add strided batch functions to hipSPARSE

### PR DESCRIPTION
**Pull request by jsandham**
Adds strided batch wrappers to hipsparse.

### CODE REVIEW

**Review by ntrost57 — APPROVED** [review]

File: clients/include/utility.hpp [hunk 1]
```diff
@@ -2554,6 +2554,270 @@ void host_csrmm(J                     M,
...
+void host_csrmm_batched(J                    M,
+                        J                    N,
+                        J                    K,
+                        J                    batch_count_A,
+                        J                    offsets_batch_stride_A,
+                        I                    columns_values_batch_stride_A,
+                        hipsparseOperation_t transA,
+                        hipsparseOperation_t transB,
+                        T                    alpha,
+                        const I*             csr_row_ptr_A,
```
Comment (YvanMokwinski) [inline clients/include/utility.hpp:2567]: __restrict__

File: clients/include/utility.hpp [hunk 2]
```diff
@@ -2554,6 +2554,270 @@ void host_csrmm(J                     M,
...
+                hipsparseIndexBase_t base)
+{
+    for(I j = 0; j < N; j++)
+    {
+#ifdef _OPENMP
+#pragma omp parallel for schedule(dynamic, 1024)
+#endif
+        for(I i = 0; i < M; ++i)
+        {
+            I idx_C  = order == HIPSPARSE_ORDER_COLUMN ? i + j * ldc : i * ldc + j;
```
Comment (YvanMokwinski) [inline clients/include/utility.hpp:2683]: would be better to put this out of the loop and having 2 loops. In one case we have an awful traversal.

**Review by YvanMokwinski — APPROVED** [review]

File: clients/include/testing_spmm_csr.hpp [hunk 3]
```diff
@@ -101,7 +101,7 @@ void testing_spmm_csr_bad_arg(void)
 
     // Create SpMM structures
     verify_hipsparse_status_success(
-        hipsparseCreateCsr(&A, m, n, nnz, dptr, dcol, dval, idxTypeI, idxTypeJ, idxBase, dataType),
```
[the remark this answers is not in the document]
Comment (jsandham) [reply clients/include/testing_spmm_csr.hpp:104]: Ya it was a bug. Fixed now.

File: clients/include/utility.hpp (hunk 2 above)
Comment (jsandham) [reply clients/include/utility.hpp]: FIxed

File: clients/include/utility.hpp [hunk 4]
```diff
@@ -2423,22 +2423,22 @@ inline void host_bsrmm(int                     Mb,
...
-                J                     ldc,
-                hipsparseOrder_t      order,
-                hipsparseIndexBase_t  base)
+void host_csrmm(J                    M,
+                J                    N,
+                J                    K,
+                hipsparseOperation_t transA,
+                hipsparseOperation_t transB,
+                T                    alpha,
+                const I*             csr_row_ptr_A,
```
[the remark this answers is not in the document]
Comment (jsandham) [reply clients/include/utility.hpp:2432]: Given that none of the other routines in this file use restrict I think ill go through and fix all of them in a separate PR. That way I am not touching 100 different unrelated utility routines in this PR.

File: clients/include/utility.hpp (hunk 1 above)
Comment (jsandham) [reply clients/include/utility.hpp:2567]: Given that none of the other routines in this file use restrict I think ill go through and fix all of them in a separate PR. That way I am not touching 100 different unrelated utility routines in this PR.

**Pull request merged**